packages/launchpad/src/components/input/Select.vue

```diff
@@ -0,0 +1,107 @@
+          <slot name="input-suffix" :value="modelValue" :open="open">
+            <Icon :icon="IconCaret" class="text-lg transform transition-transform"
+              data-testid="icon-caret"
+              aria-hidden="true"
+              :class="{
+              'rotate-0 text-indigo-600': open,
+              'rotate-180 text-gray-500': !open
+            }"/>
+          </slot>
+        </span>
+        
+      </ListboxButton>
+
+      <transition leave-active-class="transition ease-in duration-100" leave-from-class="opacity-100" leave-to-class="opacity-0">
+        <ListboxOptions class="absolute z-10 mt-1 w-full bg-white shadow-lg max-h-60 rounded-md py-1 text-base ring-1 ring-black ring-opacity-5 overflow-auto focus:outline-none sm:text-sm">
+          <ListboxOption as="ul" v-for="option in options" :key="get(option, itemKey)" :value="option" v-slot="{ active, selected }">
+              <li class="cursor-default block truncate select-none relative py-2 pl-3 pr-9" :class="[{
+                'font-medium': selected,
+                'bg-gray-50': active,
+                'text-gray-900': !active
+              }]">
+                <span class="absolute inset-y-0 flex items-center">
+                  <slot name="item-prefix" :selected="selected" :active="active" :value="option"></slot>
+                </span>
+                <span class="inline-block" :class="{
+                  'pl-8': $slots['item-prefix'],
+                  'pr-4': $slots['item-suffix'],
+                }">
+                  <slot name="item-body" :selected="selected" :active="active" :value="option">
+                    {{ get(option, itemValue) }}
+                  </slot>
+                </span>
+
+                <span class="absolute inset-y-0 right-0 pr-8 flex text-sm items-center">
+                  <slot name="item-suffix" :selected="selected" :active="active" :value="option">
+                    <span v-if="selected" class="text-indigo-500 absolute flex items-center">
+                      <Icon :icon="IconCheck"
+                        class="text-sm" 
+                        data-testid="icon-check"
+                        aria-hidden="true"/>
+                    </span>
+                  </slot>
+                </span>
+            </li>
+          </ListboxOption>
+        </ListboxOptions>
+      </transition>
+    </div>
+  </template>
+  </Listbox>
+</template>
+
+<script lang="ts" setup>
+import { Listbox, ListboxButton, ListboxLabel, ListboxOption, ListboxOptions } from '@headlessui/vue'
+import IconCheck from 'virtual:vite-icons/mdi/check'
+import IconCaret from 'virtual:vite-icons/mdi/caret'
+import Icon from '../icon/Icon.vue'
+import { get } from 'lodash'
+import { useI18n } from '../../composables'
+
+const { t } = useI18n()
+
+interface Option {
+  [key: string]: any
+}
+
+withDefaults(defineProps<{
+  options: Option[],
+  modelValue?: Option | null // Current object being selected
```

Comment:
Does this need to be `null`? It's already optional with `?` - Ideally we should only use `undefined` or `null` to represent no value, not both.